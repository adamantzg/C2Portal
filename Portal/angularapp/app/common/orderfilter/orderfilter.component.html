<div class="row">
  <div class="col-sm-12 form-inline">
    <div class="form-group">
      <label class="m-r-xs">From:</label>
      <input type="text" bs-datepicker class="form-control m-r-xs" bsDatepicker [(bsValue)]="filter.from" name="date" [bsConfig]="bsConfig" required>
    </div>
    <div class="form-group">
        <label class="m-r-xs"> To:</label>
        <input type="text" bs-datepicker class="form-control m-r-xs" bsDatepicker [(bsValue)]="filter.to" name="date" [bsConfig]="bsConfig" required>
    </div>
    <div class="form-group">
      <label class="m-r-xs">{{searchLabel}}</label>
      <input type="text" [(ngModel)]="filter.searchText" class="form-control m-r-xs">
    </div>
    <button class="btn m-l-xs" (click)="applyFilter()">Go</button>
    <button class="btn m-l-xs" (click)="clearFilter()">Clear</button>
  </div>
  <!-- <div class="col-sm-12 form-inline">
    <div class="form-group">
      <label>Filter results:</label>
      <input type="text" [(ngModel)]="filter.searchText" class="form-control">
    </div>
  </div> -->
  
</div>
