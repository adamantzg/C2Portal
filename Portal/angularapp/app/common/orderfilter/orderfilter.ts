export class OrderFilter {

    constructor(from?: Date, to?: Date, searchText: string = '') {
        this.from = from;
        this.to = to;
        this.searchText = searchText;
    }

    from: Date;
    to: Date;
    searchText = '';
}
