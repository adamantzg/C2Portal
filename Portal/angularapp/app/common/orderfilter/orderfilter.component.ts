import { Component, OnInit, Input, Output, EventEmitter} from '@angular/core';
import { OrderFilter } from './orderfilter';
import { BsDatepickerConfig } from 'ngx-bootstrap';


@Component({
  selector: 'app-orderfilter',
  templateUrl: './orderfilter.component.html',
  styleUrls: ['./orderfilter.component.css']
})
export class OrderfilterComponent implements OnInit {

  constructor() { }

  @Input()
  filter = new OrderFilter();
  bsConfig: BsDatepickerConfig = new BsDatepickerConfig();
  @Input()
  searchLabel = 'Containing text:';

  @Output()
  filterApplied = new EventEmitter();
  @Output()
  filterCleared = new EventEmitter();

  ngOnInit() {
    this.bsConfig.dateInputFormat = 'DD/MM/YYYY';
  }

  applyFilter() {
    this.filterApplied.emit();
  }

  clearFilter() {
    this.filter.searchText = '';
    this.filterCleared.emit();
  }

}
