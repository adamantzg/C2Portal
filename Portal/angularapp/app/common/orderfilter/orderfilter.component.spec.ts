import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { OrderfilterComponent } from './orderfilter.component';
import { FormsModule } from '@angular/forms';
import { BsDatepickerModule, BsDatepickerConfig, ComponentLoaderFactory, PositioningService, BsLocaleService } from 'ngx-bootstrap';

describe('OrderfilterComponent', () => {
  let component: OrderfilterComponent;
  let fixture: ComponentFixture<OrderfilterComponent>;

  setupTestBed({
    declarations: [ OrderfilterComponent ],
    imports: [FormsModule, BsDatepickerModule ],
    providers: [BsDatepickerConfig, ComponentLoaderFactory, PositioningService, BsLocaleService ]
  });


  beforeEach(() => {
    fixture = TestBed.createComponent(OrderfilterComponent);
    component = fixture.componentInstance;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  test('events', () => {
    let applied = false, cleared = false;
    component.filterApplied.subscribe(() => applied = true );
    component.filterCleared.subscribe(() => cleared = true );
    component.applyFilter();
    expect(applied).toBeTruthy();
    component.clearFilter();
    expect(cleared).toBeTruthy();
    expect(component.filter.searchText).toBe('');


  });
});
