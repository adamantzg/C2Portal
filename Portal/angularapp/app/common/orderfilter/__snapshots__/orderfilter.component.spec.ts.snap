// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderfilterComponent should create 1`] = `
<app-orderfilter
  bsConfig={[Function BsDatepickerConfig]}
  filter={[Function OrderFilter]}
  filterApplied={[Function EventEmitter]}
  filterCleared={[Function EventEmitter]}
  searchLabel={[Function String]}
>
  <div
    class="row"
  >
    <div
      class="col-sm-12 form-inline"
    >
      <div
        class="form-group"
      >
        <label>
          From:
        </label>
        <input
          bs-datepicker=""
          bsdatepicker=""
          class="form-control"
          name="date"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
           To:
        </label>
        <input
          bs-datepicker=""
          bsdatepicker=""
          class="form-control"
          name="date"
          required=""
          type="text"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          
        </label>
        <input
          class="form-control"
          type="text"
        />
      </div>
      <button
        class="btn"
      >
        Go
      </button>
      <button
        class="btn"
      >
        Clear
      </button>
    </div>
  </div>
</app-orderfilter>
`;
